okhttp/src/jvmTest/java/okhttp3/OkHttpClientTest.java

```diff
@@ -1,405 +0,0 @@
-    } catch (IllegalArgumentException ignored) {
-    }
-    try {
-      builder.readTimeout(Duration.ofNanos(1));
-    } catch (IllegalArgumentException ignored) {
-    }
-    try {
-      builder.callTimeout(Duration.ofDays(365));
-    } catch (IllegalArgumentException ignored) {
-    }
-    try {
-      builder.connectTimeout(Duration.ofDays(365));
-    } catch (IllegalArgumentException ignored) {
-    }
-    try {
-      builder.writeTimeout(Duration.ofDays(365));
-    } catch (IllegalArgumentException ignored) {
-    }
-    try {
-      builder.readTimeout(Duration.ofDays(365));
-    } catch (IllegalArgumentException ignored) {
-    }
-  }
-
-  @Test public void clonedInterceptorsListsAreIndependent() throws Exception {
-    Interceptor interceptor = chain -> chain.proceed(chain.request());
-    OkHttpClient original = clientTestRule.newClient();
-    original.newBuilder()
-        .addInterceptor(interceptor)
-        .addNetworkInterceptor(interceptor)
-        .build();
-    assertThat(original.interceptors().size()).isEqualTo(0);
-    assertThat(original.networkInterceptors().size()).isEqualTo(0);
-  }
-
-  /**
-   * When copying the client, stateful things like the connection pool are shared across all
-   * clients.
-   */
-  @Test public void cloneSharesStatefulInstances() throws Exception {
-    OkHttpClient client = clientTestRule.newClient();
-
-    // Values should be non-null.
-    OkHttpClient a = client.newBuilder().build();
-    assertThat(a.dispatcher()).isNotNull();
-    assertThat(a.connectionPool()).isNotNull();
-    assertThat(a.sslSocketFactory()).isNotNull();
-    assertThat(a.x509TrustManager()).isNotNull();
-
-    // Multiple clients share the instances.
-    OkHttpClient b = client.newBuilder().build();
-    assertThat(b.dispatcher()).isSameAs(a.dispatcher());
-    assertThat(b.connectionPool()).isSameAs(a.connectionPool());
-    assertThat(b.sslSocketFactory()).isSameAs(a.sslSocketFactory());
-    assertThat(b.x509TrustManager()).isSameAs(a.x509TrustManager());
-  }
-
-  @Test public void setProtocolsRejectsHttp10() throws Exception {
-    OkHttpClient.Builder builder = new OkHttpClient.Builder();
-    try {
-      builder.protocols(asList(Protocol.HTTP_1_0, Protocol.HTTP_1_1));
-      fail();
-    } catch (IllegalArgumentException expected) {
-    }
-  }
-
-  @Test public void certificatePinnerEquality() {
-    OkHttpClient clientA = clientTestRule.newClient();
-    OkHttpClient clientB = clientTestRule.newClient();
-    assertThat(clientB.certificatePinner()).isEqualTo(clientA.certificatePinner());
-  }
-
-  @Test public void nullInterceptor() {
-    OkHttpClient.Builder builder = new OkHttpClient.Builder();
-    try {
-      builder.addInterceptor(null);
-      fail();
-    } catch (NullPointerException expected) {
-    }
-  }
-
-  @Test public void nullNetworkInterceptor() {
-    OkHttpClient.Builder builder = new OkHttpClient.Builder();
-    try {
-      builder.addNetworkInterceptor(null);
```

Comment:
I removed this test (and a couple similar) because Kotlin doesn’t allow us to express it. We’re letting the type system do something Java required us to do manually.